strings-csharp/HexToByteArray/HexToByteArray/ConvertHex.cs

```diff
@@ -16,5 +18,157 @@ public static byte[] FromHexStringAlternative(ReadOnlySpan<char> hex)
                     System.Globalization.NumberStyles.HexNumber);
             return bytes;
         }
+
+        public static byte[] FromHexWithPointers(string hex)
```

Comment:
This method is not using pointers, it is performing a modular arithmetic calculation. So it would be better to rename it to something like FromHexWithModularArithmetic, or something along those lines. Note, this is one of the ones I mentioned in the review document and have in my PR with the calculation cleaned up and a method extracted to make the code a little more readable (if you notice, the calculation for both firstChar and secondChar are identical). You could just copy the method directly from my PR.

Reply:
Just as a follow-up on the name change to this method. In C# the idea of a pointer is very specific, and their use requires an unsafe code block. We can do something similar without unsafe in more modern C# using spans, references and the Unsafe methods, but even then, we should not use the word pointer as a pointer and a reference are different things. I tried writing this method using all three options and found that there was almost no performance difference since most of the time in this method is spent in the calculation and not in the retrieving of data from the source array and copying to the destination array.